modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp

```diff
@@ -23,6 +23,12 @@ DLAdminManagementToolbarDisplayContext dlAdminManagementToolbarDisplayContext =
 DLViewDisplayContext dlViewDisplayContext = new DLViewDisplayContext(dlAdminDisplayContext, request, renderRequest, renderResponse);
 %>
 
+<liferay-frontend:stylesheet bundle="com.liferay.frontend.taglib" css="example.css" />
```

Comment:
Here, API contract is: load the style which can be found in the provider bundle, given
* `css` is a local path interpreted in the provider bundle relative to `/META-INF/resources/` 
* `bundle` is the symbolic name as declared in the `Bundle-SymbolicName` header in the MANIFEST.MF of the provider bundle

So, any bundle can be considered as CSS provider as long as:
* It's active in the OSGi runtime at the moment tag is run
* It includes css resources somewhere in `src/main/resources/META-INF/resources` and 
* Its `Bundle-SymbolicName` header in `bnd.bnd` is known to the JSP

Is this correct?

Reply:
Maybe the provider module needs also the ability to serve the css on its own (i.e. a web context path)?

Reply:
The provider needs (with this implementation):

1. A `Web-ContextPath` (to serve the CSS files)
2. A build based on `liferay-npm-scripts` or `@liferay/cli` (not yet released). Otherwise, the user needs to configure the bundler and use any of the styles loaders that make CSS available as AMD modules.

Reply:
However, we could expand the implementation of the taglib so that, if the bundle:

1. Doesn't comply with 1 above: we serve the CSS through an auxiliary `Web-ContextPath` (for example the one from `liferay-frontend-taglib`, or a different new bundle we create).
2. Doesn't fulfill 2 above: we try to load the CSS by other means (for example: simply printing a `<link>` element, rendering the file inline, writing some ad-hoc JS to load the CSS just once, or something along those lines...

The benefit of using the AMD loader is that, if JS code tries to load the CSS again, it will not duplicate the request, because everything is orchestrated by the AMD loader. If we implement these alternatives, we will need to either allow duplication of CSS resources or couple these features and the AMD loader so that they cooperate to deduplicate CSS requests.